sympy/physics/control/lti.py

```diff
@@ -1,14 +1,24 @@
-from sympy import Basic, Mul, degree, Symbol, sympify
+from sympy import Basic, Mul, degree, Symbol, sympify, expand
+from sympy.core.numbers import Integer
 
 __all__ = ['TransferFunction',]
 
 
-class TransferFunction(Mul):
-    # this will subclass from `Basic`.
-    def __new__(cls, num, den):
-        obj = Mul.__new__(cls, num, 1/den)
-        obj.num = num
-        obj.den = den
+class TransferFunction(Basic):
+
+    def __new__(cls, num, den, var):
+
+        if not isinstance(var, Symbol):
+            raise TypeError("Last argument must be a complex Symbol.")
```

Comment:
It's only testing if var is a Symbol.
There should be a different Error message here

Reply:
Also, you need to add a different test to check the variable is complex.

Reply:
> Also, you need to add a different test to check the variable is complex.

By test you mean unit test? I will add a test like this: `raises(TypeError, lambda: TransferFunction(s - 1, s + 2, 3))`

Reply:
If `var` is a `Symbol`, it should raise an error if `var.is_real` is `True`, to check whether the variable is complex.

Reply:
I get this: 
``` 
>>> x = Symbol('x', real=True)
>>> x.is_complex
True
```

Reply:
@oscarbenjamin, is that a bug?

Reply:
This shouldn't be happening. I also tested it a bit. The other property remains `None` in all the cases except when `real = True`. 

```python 
>>> x = Symbol('x')
>>> y = Symbol('y', real = True)
>>> z = Symbol('z', real = False)
>>> p = Symbol('p', complex = True)
>>> x.is_real
>>> x.is_complex
>>> y.is_real
True
>>> y.is_complex
True
>>> z.is_complex
>>> z.is_real
False
>>> p.is_real
>>> p.is_complex
True
```

Reply:
The reals are a subset of the complex numbers

Reply:
Then checking for `var.is_real` is not a good idea. Perhaps, we should test for `not var.is_complex` then.

Reply:
> we should test for `not var.is_complex` then.

Yes, that should work.

Reply:
Is there any use of restricting the symbol to be complex?
For sympy's polynomial module, I don't think that it tests whether the symbols are real or complex, even if the roots can be computed in complex numbers.

Reply:
Yeah, I also think the only criterion must be that the `var` is a Symbol. 
Should I store `num` & `den` as `Poly()` objects internally? 
I see it becomes the output becomes totally unreadable if we want to keep the numerator and the denominator in non expanded form. 
If not, then how do I check that the user doesn't input anything other than polynomials or integers. One way could be using `Expr`:
``` python
(Pdb) isinstance(p**2 - p*s + 1, Expr)
True
```
But that could also include other arithmetic operations..